vms/platformvm/txs/builder/builder.go

```diff
@@ -28,369 +26,108 @@ import (
-		outOwner,
-		options(changeAddr, memo)...,
+		chainID,
+		to,
+		options...,
 	)
 	if err != nil {
 		return nil, fmt.Errorf("failed building import tx: %w", err)
 	}
 
-	tx, err := walletsigner.SignUnsigned(context.Background(), pSigner, utx)
-	if err != nil {
-		return nil, err
-	}
-	return tx, tx.SyntacticVerify(b.ctx)
+	return walletsigner.SignUnsigned(context.Background(), pSigner, utx)
 }
 
-// TODO: should support other assets than AVAX
-func (b *builder) NewExportTx(
-	amount uint64,
+func (b *Builder) NewExportTx(
 	chainID ids.ID,
-	to ids.ShortID,
+	outputs []*avax.TransferableOutput,
 	keys []*secp256k1.PrivateKey,
-	changeAddr ids.ShortID,
-	memo []byte,
+	options ...common.Option,
 ) (*txs.Tx, error) {
 	pBuilder, pSigner := b.builders(keys)
 
-	outputs := []*avax.TransferableOutput{{
-		Asset: avax.Asset{ID: b.ctx.AVAXAssetID},
-		Out: &secp256k1fx.TransferOutput{
-			Amt: amount,
-			OutputOwners: secp256k1fx.OutputOwners{
-				Locktime:  0,
-				Threshold: 1,
-				Addrs:     []ids.ShortID{to},
-			},
-		},
-	}}
-
 	utx, err := pBuilder.NewExportTx(
 		chainID,
 		outputs,
-		options(changeAddr, memo)...,
+		options...,
 	)
 	if err != nil {
 		return nil, fmt.Errorf("failed building export tx: %w", err)
 	}
 
-	tx, err := walletsigner.SignUnsigned(context.Background(), pSigner, utx)
-	if err != nil {
-		return nil, err
-	}
-	return tx, tx.SyntacticVerify(b.ctx)
+	return walletsigner.SignUnsigned(context.Background(), pSigner, utx)
 }
 
-func (b *builder) NewCreateChainTx(
+func (b *Builder) NewCreateChainTx(
 	subnetID ids.ID,
-	genesisData []byte,
+	genesis []byte,
 	vmID ids.ID,
 	fxIDs []ids.ID,
 	chainName string,
 	keys []*secp256k1.PrivateKey,
-	changeAddr ids.ShortID,
-	memo []byte,
+	options ...common.Option,
 ) (*txs.Tx, error) {
 	pBuilder, pSigner := b.builders(keys)
 
-	utx, err := pBuilder.NewCreateChainTx(subnetID, genesisData, vmID, fxIDs, chainName, options(changeAddr, memo)...)
+	utx, err := pBuilder.NewCreateChainTx(
+		subnetID,
+		genesis,
+		vmID,
+		fxIDs,
+		chainName,
+		options...,
+	)
 	if err != nil {
 		return nil, fmt.Errorf("failed building create chain tx: %w", err)
 	}
 
-	tx, err := walletsigner.SignUnsigned(context.Background(), pSigner, utx)
-	if err != nil {
-		return nil, err
-	}
-	return tx, tx.SyntacticVerify(b.ctx)
+	return walletsigner.SignUnsigned(context.Background(), pSigner, utx)
 }
 
-func (b *builder) NewCreateSubnetTx(
-	threshold uint32,
-	ownerAddrs []ids.ShortID,
+func (b *Builder) NewCreateSubnetTx(
+	owner *secp256k1fx.OutputOwners,
 	keys []*secp256k1.PrivateKey,
-	changeAddr ids.ShortID,
-	memo []byte,
+	options ...common.Option,
 ) (*txs.Tx, error) {
 	pBuilder, pSigner := b.builders(keys)
 
-	utils.Sort(ownerAddrs) // sort control addresses
-	subnetOwner := &secp256k1fx.OutputOwners{
-		Threshold: threshold,
-		Addrs:     ownerAddrs,
-	}
-
-	utx, err := pBuilder.NewCreateSubnetTx(subnetOwner, options(changeAddr, memo)...)
+	utx, err := pBuilder.NewCreateSubnetTx(
```

Comment:
I believe we should keep `utils.Sort(owner.Addrs)` here.
All UTs passes here, but I have others down the line that relays on the sort.
Moreover when we verify the transactions we do check that owner addresses are duly ordered.